api/filters/shared_instance_update_filter.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filters
+
+import (
+	"encoding/json"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+	"github.com/sirupsen/logrus"
+	"net/http"
+)
+
+const SharingInstanceUpdateFilterName = "SharedInstanceUpdateFilter"
+
+// SharingInstanceFilter validate the un/share request on an existing service instance
+type sharedInstanceUpdateFilter struct {
+	storageRepository storage.Repository
+}
+
+func NewSharedInstanceUpdateFilter(storageRepository storage.Repository) *sharedInstanceUpdateFilter {
+	return &sharedInstanceUpdateFilter{
+		storageRepository: storageRepository,
+	}
+}
+
+func (*sharedInstanceUpdateFilter) Name() string {
+	return SharingInstanceUpdateFilterName
+}
+
+func (sf *sharedInstanceUpdateFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
+	var reqServiceInstance types.ServiceInstance
+	err := util.BytesToObjectNoLabels(req.Body, &reqServiceInstance)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := req.Context()
+	instanceID := req.PathParams["resource_id"]
+
+	// Get instance from database
+	dbPersistedInstanceObject, err := storage.GetObjectByField(ctx, sf.storageRepository, types.ServiceInstanceType, "id", instanceID)
```

Comment:
and another thing about this query.(and probably also to other queries in these flows) - we are not using the context criteria in the queries here, couldn't it lead to issues? e.g. breaking tenant separation, container isolation etc.

Reply:
Added to the GetObjectByField a new condition for adding the `CriteriaForContext` to the query for service-instance types.